app/src/main/java/com/example/worldnews/domain/usecase/NewsUseCase.kt

```diff
@@ -0,0 +1,39 @@
+package com.example.worldnews.domain.usecase
+
+import com.example.worldnews.common.Resource
+import com.example.worldnews.domain.model.News
+import kotlinx.coroutines.Dispatchers.IO
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.flow.flowOn
+import retrofit2.HttpException
+import java.io.IOException
+import javax.inject.Inject
+
+open class NewsUseCase @Inject constructor(
+    private val news: suspend (
+        category: String,
```

Comment:
wrap the inputs into a data class

Reply:
is there any boilerplate code reamin in children or its all good?

Reply:
It's just a rule of clean code (not too many inputs)